### PR TITLE
Security Fix for alm dependency

### PR DESCRIPTION
**Pull request by manas-sf**
In order to remove dependency of **salesforce-alm** 
We took the approach of executing the commands over shell. 
This has pre-requisite that  sf cli should be installed where vbt is executing otherwise it will fail. 
Also sf cli should either be **logged in** with the process where VBT commands are getting executed or **authFilePath** should be provided as input to the VBT . 
This authFile can be obtained by running org display in Already logged in cli. You can use web login to get the Auth File.  Refer - https://developer.salesforce.com/docs/atlas.en-us.sfdx_cli_reference.meta/sfdx_cli_reference/cli_reference_org_commands_unified.htm#cli_reference_org_login_sfdx-url_unified

https://developer.salesforce.com/docs/atlas.en-us.sfdx_cli_reference.meta/sfdx_cli_reference/cli_reference_org_commands_unified.htm#cli_reference_org_login_web_unified

### CODE REVIEW

File: lib/datapacksutils.js [hunk 1]
```diff
@@ -4861,6 +4862,21 @@ DataPacksUtils.prototype.formatSalesforceDeployResponse = function (response) {
...
 }
 
+DataPacksUtils.prototype.formatSalesforceDeployResponseFromCLI = function (response) {
+    VlocityUtils.report("Source deploy response is ",response);
+    let currentStatusObj = {};
+    
+    if (response.details.componentSuccesses) {
+        response = response.details.componentSuccesses;
+    }
+     for( component of response){
```
Comment (manas-sf) [inline lib/datapacksutils.js:4872]: Will response be always an array? What if any exception or error occurs?
Above if can be simply `response = response?.details?.componentSuccesses;`

File: lib/utilityservice.js [hunk 2]
```diff
@@ -343,7 +343,8 @@ UtilityService.prototype.sfdxLogin = async function() {
     VlocityUtils.report('Refreshing SFDX Session', this.vlocity.sfdxUsername);
 
     try {
-        var orgInfo = await this.sfdx('org:display', { targetusername: this.vlocity.sfdxUsername, verbose: true });
+        var orgInfo = await this.sfdx('org display', { targetusername: this.vlocity.sfdxUsername, verbose: true });
+        VlocityUtils.report("SFDX login sucessful for user ", orgInfo.username);
```
Comment (manas-sf) [inline lib/utilityservice.js:347]: typo: successful

File: lib/utilityservice.js [hunk 3]
```diff
@@ -368,6 +369,9 @@ UtilityService.prototype.checkLogin = async function() {
     
     VlocityUtils.verbose('Check Login');
     try {
+        if(this.vlocity.authFilePath){
```
Comment (manas-sf) [inline lib/utilityservice.js:372]: indentation missing in this block

File: lib/utilityservice.js [hunk 4]
```diff
@@ -410,6 +414,45 @@ UtilityService.prototype.checkLogin = async function() {
...
+    };
+
+    var flags = options;
+
+    flags.quiet = VlocityUtils.quiet;
+    flags.json = VlocityUtils.quiet || !VlocityUtils.showLoggingStatements;
+
+    let allCacheKeys = Object.keys(require.cache);
+
+    for (let key of allCacheKeys) {
```
Comment (manas-sf) [inline lib/utilityservice.js:434]: What is this for? what cache are we suspecting?

File: lib/utilityservice.js [hunk 5]
```diff
@@ -853,90 +896,14 @@ UtilityService.prototype.getVlocitySetting = async function(settingName) {
 }
 
 UtilityService.prototype.getSfdxCommandRunner = function(jobInfo) {
-    if(jobInfo.useSfdxCli === true && commandExistsSync('sfdx')) {
+    if(!commandExistsSync('sf')){
```
Comment (manas-sf) [inline lib/utilityservice.js:899]: indentation

File: lib/utilityservice.js (hunk 4 above)
Comment (AbhinavKumar-sf) [reply lib/utilityservice.js:434]: It was there with older method also , i guess  this can be removed, removing it

File: lib/utilityservice.js [hunk 6]
```diff
@@ -853,90 +896,14 @@ UtilityService.prototype.getVlocitySetting = async function(settingName) {
...
+        VlocityUtils.log(' Cannot  proceed further Sf command doesnot exist , please install sf');
```
Comment (manas-sf) [inline lib/utilityservice.js:900]: Please install SFDX as tool can't proceed further without sf commands.

This should be VlocityUtils.error  or something more appropriate.

**Review by manas-sf — COMMENTED** [review]
@AbhinavKumar-sf Please add description to PR as what will work and what will not work post these changes?

File: lib/utilityservice.js (hunk 4 above)
Comment (manas-sf) [reply lib/utilityservice.js:434]: If it was there, Let's not remove it. We are not sure of consequences.

File: lib/datapacksutils.js (hunk 1 above)
Comment (AbhinavKumar-sf) [reply lib/datapacksutils.js]: This change has been accommodated

File: lib/utilityservice.js (hunk 2 above)
Comment (AbhinavKumar-sf) [reply lib/utilityservice.js]: This change has been accommodated

File: lib/utilityservice.js (hunk 3 above)
Comment (AbhinavKumar-sf) [reply lib/utilityservice.js]: This change has been accommodated

File: lib/utilityservice.js (hunk 4 above)
Comment (AbhinavKumar-sf) [reply lib/utilityservice.js:434]: removed it tests are passing, for now anyways we are executing things on shell, this cache will not have any impact

File: lib/utilityservice.js (hunk 5 above)
Comment (AbhinavKumar-sf) [reply lib/utilityservice.js]: This change has been accommodated

File: lib/utilityservice.js (hunk 6 above)
Comment (AbhinavKumar-sf) [reply lib/utilityservice.js]: This change has been accommodated

**Pull request merged**